.github/workflows/scripts/libcommon.sh

```diff
@@ -0,0 +1,144 @@
+        wget https://github.com/mozilla/sccache/releases/download/v0.2.15/sccache-v0.2.15-aarch64-apple-darwin.tar.gz
+        tar -xzf sccache-v0.2.15-aarch64-apple-darwin.tar.gz
+        chmod +x sccache-v0.2.15-aarch64-apple-darwin/sccache
+        export PATH=$(pwd)/sccache-v0.2.15-aarch64-apple-darwin:$PATH
+    else
+        wget https://github.com/mozilla/sccache/releases/download/v0.2.15/sccache-v0.2.15-x86_64-apple-darwin.tar.gz
+        tar -xzf sccache-v0.2.15-x86_64-apple-darwin.tar.gz
+        chmod +x sccache-v0.2.15-x86_64-apple-darwin/sccache
+        export PATH=$(pwd)/sccache-v0.2.15-x86_64-apple-darwin:$PATH
+    fi
+}
+
+setup_python() {
+    if [[ "$IN_DOCKER" == "true" ]]; then
+        source $HOME/miniconda/etc/profile.d/conda.sh
+        conda activate "$PY"
+    fi
+    python3 -m pip uninstall taichi taichi-nightly -y
+    python3 -m pip install -r requirements_dev.txt
+}
+
+function setup-sccache-local {
+    if [ -z $SCCACHE_ROOT ]; then
+        echo "Skipping sccache setup since SCCACHE_ROOT is not set"
+        return
+    fi
+
+    mkdir -p $SCCACHE_ROOT/{bin,cache}
+
+    export SCCACHE_DIR=$SCCACHE_ROOT/cache
+    export SCCACHE_CACHE_SIZE="10G"
+    export SCCACHE_LOG=error
+    export SCCACHE_ERROR_LOG=$SCCACHE_ROOT/sccache_error.log
+
+    echo "SCCACHE_ROOT: $SCCACHE_ROOT"
+
+    if [ ! -x $SCCACHE_ROOT/bin/sccache ]; then
+        if [[ $OSTYPE == "linux-"* ]]; then
+            wget https://github.com/mozilla/sccache/releases/download/v0.2.15/sccache-v0.2.15-x86_64-unknown-linux-musl.tar.gz
+            tar -xzf sccache-v0.2.15-x86_64-unknown-linux-musl.tar.gz
+            mv sccache-v0.2.15-x86_64-unknown-linux-musl/* $SCCACHE_ROOT/bin
+        elif [[ $(uname -m) == "arm64" ]]; then
+            wget https://github.com/mozilla/sccache/releases/download/v0.2.15/sccache-v0.2.15-aarch64-apple-darwin.tar.gz
+            tar -xzf sccache-v0.2.15-aarch64-apple-darwin.tar.gz
+            mv sccache-v0.2.15-aarch64-apple-darwin/* $SCCACHE_ROOT/bin
+        else
+            wget https://github.com/mozilla/sccache/releases/download/v0.2.15/sccache-v0.2.15-x86_64-apple-darwin.tar.gz
+            tar -xzf sccache-v0.2.15-x86_64-apple-darwin.tar.gz
+            mv sccache-v0.2.15-x86_64-apple-darwin/* $SCCACHE_ROOT/bin
+        fi
+        chmod +x $SCCACHE_ROOT/bin/sccache
+    fi
+
+    export PATH=$SCCACHE_ROOT/bin:$PATH
+    export TAICHI_CMAKE_ARGS="$TAICHI_CMAKE_ARGS -DCMAKE_C_COMPILER_LAUNCHER=sccache -DCMAKE_CXX_COMPILER_LAUNCHER=sccache"
+}
+
+function setup-android-ndk-env {
+    export ANDROID_NDK_ROOT=/android-sdk/ndk-bundle
+    export ANDROID_CMAKE_ARGS="-DCMAKE_TOOLCHAIN_FILE=${ANDROID_NDK_ROOT}/build/cmake/android.toolchain.cmake -DANDROID_NATIVE_API_LEVEL=29 -DANDROID_ABI=arm64-v8a"
+    export TAICHI_CMAKE_ARGS="$TAICHI_CMAKE_ARGS $ANDROID_CMAKE_ARGS"
+    export PATH=$ANDROID_NDK_ROOT/toolchains/llvm/prebuilt/linux-x86_64/bin:$PATH
+}
+
+function grab-android-bot {
```

Comment:
do we still need this function?

Reply:
Yes, is part of Android smoke test process.
This function is used for grabbing an idle Android phone to install built apks, pairing with `release-android-bot` function for releasing.